sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/util/ResolveDefaultColumnsUtil.scala

```diff
@@ -376,7 +376,7 @@ object ResolveDefaultColumns extends QueryErrorsBase
     val expr = Literal.fromSQL(defaultSQL) match {
       // EXISTS_DEFAULT will have a cast from analyze() due to coerceDefaultValue
```

Comment:
Don't we expect the cast to be evaluated before the exists default is persisted?

Reply:
Yes in the happy path.

There are unfortunately bugs from before though, where column exists_defaults were persisted as unresolved, hence the fallback mechanism.